src/Compilers/CSharp/Portable/FlowAnalysis/PreciseAbstractFlowPass.cs

```diff
@@ -2733,6 +2733,41 @@ public override BoundNode VisitDiscardExpression(BoundDiscardExpression node)
         {
             return null;
         }
+
+        public override BoundNode VisitConstructorMethodBody(BoundConstructorMethodBody node)
+        {
+            Visit(node.Initializer);
+            VisitMethodBodies(node.BlockBody, node.ExpressionBody);
+            return null;
+        }
+
+        public override BoundNode VisitNonConstructorMethodBody(BoundNonConstructorMethodBody node)
+        {
+            VisitMethodBodies(node.BlockBody, node.ExpressionBody);
+            return null;
+        }
+
+        private void VisitMethodBodies(BoundBlock blockBody, BoundBlock expressionBody)
+        { 
+            if (blockBody == null)
+            {
+                Visit(expressionBody);
+                return;
+            }
+            else if (expressionBody == null)
+            {
+                Visit(blockBody);
+                return;
+            }
+
+            LocalState initialState = this.State.Clone();
+            Visit(blockBody);
+            LocalState afterBlock = this.State;
+            SetState(initialState);
+            Visit(expressionBody);
+
+            IntersectWith(ref this.State, ref afterBlock);
+        }
```

Comment:
The intersection implies there is a merge of control-flow paths being simulated here. Please explain the semantics of the control-flow that are being simulated and the rationale.

Reply:
> The intersection implies there is a merge of control-flow paths being simulated here. Please explain the semantics of the control-flow that are being simulated and the rationale.

As discussed offline, I will add the explanation as a comment.

---
In reply to: [170120529](https://github.com/dotnet/roslyn/pull/24918#discussion_r170120529) [](ancestors = 170120529)